wgsl/extract-grammar.py

```diff
@@ -510,7 +761,321 @@ def read_spec(options):
+                    if bnfdialect_data[s_j] in RULE_NAME_CODE_POINTS:
+                        rule_value += bnfdialect_data[s_j]
+                        s_j += 1
+                    elif bnfdialect_data[s_j] == ":":
+                        is_rule = True
+                        s_j += 1
+                        break
+                    # Return false when line or rule ends, pattern or token starts
+                    elif bnfdialect_data[s_j] in ["\n", ";", "/", "'"]:
+                        is_rule = False
+                        break
+                    else:
+                        s_j += 1
+                if is_rule:
+                    scan_in["name"] = "rule"
+                    scan_in["value"] = rule_value
+                    scan_in["content"] = [0, 0]
+                    s_i = s_j
+                    syntax_dict[rule_value] = {
+                        "type": "",
+                        "members": [{"type": "sequence", "members": []}],
+                    }
+                else:
+                    s_i += 1
+            else:
+                s_i += 1
+        elif scan_in["name"] == "comment":
+            # Comment nesting start
+            if s_v == "/" and s_i + 1 < len(bnfdialect_data):
+                if bnfdialect_data[s_i + 1] == "*":
+                    scan_in["value"] += 1
+                    s_i += 2
+                else:
+                    s_i += 1
+            # Comment or comment nesting end
+            elif s_v == "*" and s_i + 1 < len(bnfdialect_data):
+                if bnfdialect_data[s_i + 1] == "/":
+                    scan_in["value"] -= 1
+                    if scan_in["value"] == 0:
+                        if len(scan_in["content"]) == 0:
+                            scan_in["name"] = "blank"
+                            scan_in["value"] = ""
+                            scan_in["content"] = []
+                        else:
+                            scan_in = scan_in["content"][-1]
+                    s_i += 2
+                else:
+                    s_i += 1
+            else:
+                s_i += 1
+        elif scan_in["name"] == "rule":
+            # Comment or pattern start
+            if s_v == "/" and s_i + 1 < len(bnfdialect_data):
+                if bnfdialect_data[s_i + 1] == "*":
+                    scan_in["name"] = "comment"
+                    scan_in["value"] = 1
+                    scan_in["content"] = [scan_in]
+                    s_i += 2
+                else:  # Pattern literal
+                    pattern_value = "/"
+                    s_j = s_i + 1
+                    while s_j < len(bnfdialect_data):
+                        pattern_value += bnfdialect_data[s_j]
+                        if bnfdialect_data[s_j] == "/":
+                            s_j += 1
+                            while s_j < len(bnfdialect_data):
+                                if bnfdialect_data[s_j] in string.ascii_lowercase:
```

Comment:
`y` flag is from JavaScript, and not including it can easily lead people to use RegEx while parsing a subset than entire identifier. I suggest to preserve both `u` and `y` for the RegEx. I'll address other comments a bit later!